bgpd/bgp_rpki.c

```diff
@@ -941,16 +1049,25 @@ static int config_write(struct vty *vty)
 {
 	struct listnode *cache_node;
 	struct cache *cache;
+	struct rpki_vrf *rpki_vrf;
+	struct list *cache_list;
 
-	if (config_changed()) {
+	rpki_vrf = find_rpki_vrf(NULL);
+	if (!rpki_vrf)
+		return ERROR;
+	if (config_changed(rpki_vrf)) {
 		vty_out(vty, "!\n");
 		vty_out(vty, "rpki\n");
-		if (polling_period != POLLING_PERIOD_DEFAULT)
-			vty_out(vty, "  rpki polling_period %d\n", polling_period);
-		if (retry_interval != RETRY_INTERVAL_DEFAULT)
-			vty_out(vty, "  rpki retry-interval %d\n", retry_interval);
-		if (expire_interval != EXPIRE_INTERVAL_DEFAULT)
-			vty_out(vty, "  rpki expire_interval %d\n", expire_interval);
+		if (rpki_vrf->polling_period != POLLING_PERIOD_DEFAULT)
```

Comment:
we are doing the same check already in config_changed() function.

Reply:
yes, but it is global:
- I mean, first you check if it is worth dumping config
- then when you know there is config, you check which line to dump. 

what do you mean, please?